engine/src/board.rs

```diff
@@ -61,6 +127,141 @@ impl Board {
         }
     }
 
+    pub fn create_svg(&self, mut path: PathBuf) -> Result<()> {
+        path.set_extension("svg");
+        let file = OpenOptions::new()
+            .read(true)
+            .write(true) // Required for creation
+            .create(true)
+            .truncate(true)
+            .open(&path)?;
+        let mut writer = BufWriter::new(file);
+
+        let (mut min_x, mut max_x, mut min_y, mut max_y) = (f32::MAX, f32::MIN, f32::MAX, f32::MIN);
+        let mut positions_piece = Vec::new();
+
+        for (offset, maybe_position) in self.positions.iter().enumerate() {
+            if let Some(position) = maybe_position {
+                let piece = Self::offset_to_piece(offset);
+                let level = self.level_of_piece(piece, *position).unwrap_or(0);
+                let center = SvgPosition::center_for_level(*position, level, true);
+                if center.0 < min_x {
+                    min_x = center.0;
+                }
+                if center.0 > max_x {
+                    max_x = center.0;
+                }
+                if center.1 < min_y {
+                    min_y = center.1;
+                }
+                if center.1 > max_y {
+                    max_y = center.1;
+                }
+                positions_piece.push((position, piece));
+            }
+        }
+
+        positions_piece.sort_by(|(pos_a, piece_a), (pos_b, piece_b)| {
+            let level_a = self.level_of_piece(*piece_a, **pos_a).unwrap_or(0);
+            let level_b = self.level_of_piece(*piece_b, **pos_b).unwrap_or(0);
+            if level_a != level_b {
+                level_a.cmp(&level_b)
+            } else if pos_a.r != pos_b.r {
+                pos_a.r.cmp(&pos_b.r)
+            } else {
+                pos_a.q.cmp(&pos_b.q)
+            }
+        });
+
+        let piece_height = 104.242;
+        let piece_width = 88.338;
+        let space_around = 20.0;
+        let width = max_x - min_x + piece_width + 2.0 * space_around;
+        let height = max_y - min_y + piece_height + 2.0 * space_around;
+
+        writeln!(
+            writer,
+            "<svg width=\"{width}\" height=\"{height}\" xmlns=\"http://www.w3.org/2000/svg\">"
+        )?;
+        let pieces = std::fs::read_to_string("pieces.svg")?;
+        writeln!(writer, "{pieces}")?;
+        writeln!(
+            writer,
+            "<g transform=\"translate({} {})\">",
+            -1.0 * min_x + space_around,
+            -1.0 * min_y + space_around
+        )?;
+
+        for (position, piece) in positions_piece.iter() {
```

Comment:
This needs to be turned into an impl Iter on board so it can be reused from other places